`index.d.ts` contains Vue 2 type references

## Problem

The published `index.d.ts` imports Vue 2-only APIs that do not exist in Vue 3:

```ts
import Vue, { PluginFunction } from "vue"
import { DirectiveOptions } from 'vue/types/options'
```

- `PluginFunction` was removed in Vue 3 (replaced by `Plugin` / `ObjectPlugin`)
- `vue/types/options` does not exist in Vue 3
- Vue 3 does not have a default export from `"vue"`

Any project that disables `skipLibCheck` in `tsconfig.json` will get three compiler errors from this file:

```
node_modules/vue3-observe-visibility/index.d.ts(1,8): error TS1192: Module '"vue"' has no default export.
node_modules/vue3-observe-visibility/index.d.ts(1,15): error TS2305: Module '"vue"' has no exported member 'PluginFunction'.
node_modules/vue3-observe-visibility/index.d.ts(2,34): error TS2307: Cannot find module 'vue/types/options' or its corresponding type declarations.
```

## Wvaluation

The declaration file was written for `vue-observe-visibility` (the Vue 2 predecessor) and was not updated when the package was ported to Vue 3. The runtime code is correct Vue 3; only the type declarations are wrong.

## Suggested fix

Replace `index.d.ts` with declarations that use the Vue 3 type API:

```ts
import type { Directive, Plugin } from 'vue';

export declare const ObserveVisibility: Directive;
declare const plugin: Plugin;
export default plugin;
```

This matches the actual runtime:

- The default export is `{ install(app) { app.directive('observe-visibility', ObserveVisibility) } }`, correctly typed as `Plugin`
- The named export `ObserveVisibility` is a directive object with `beforeMount`, `updated`, `unmounted`, and `getSSRProps` hooks, correctly typed as `Directive`

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,8 +1,5 @@
-import Vue, { PluginFunction } from "vue"
-import { DirectiveOptions } from 'vue/types/options'
+import type { Directive, Plugin } from 'vue';
 
-export const ObserveVisibility: DirectiveOptions
-
-export default class VueObserveVisibilityPlugin {
-	static install: PluginFunction<never>
-}
+export declare const ObserveVisibility: Directive;
+declare const plugin: Plugin;
+export default plugin;

